shared/src/business/useCases/editDocketEntry/completeDocketEntryQCInteractor.js

```diff
@@ -41,45 +49,93 @@ exports.completeDocketEntryQCInteractor = async ({
     });
 
   const caseEntity = new Case(caseToUpdate, { applicationContext });
+  const { index: docketRecordIndexUpdated } = caseEntity.docketRecord.find(
+    record => record.documentId === documentId,
+  );
 
   const currentDocument = caseEntity.getDocumentById({
     documentId,
   });
 
-  const needsNewCoversheet =
-    entryMetadata.additionalInfo != currentDocument.additionalInfo ||
-    entryMetadata.documentTitle != currentDocument.documentTitle;
-
-  const documentEntity = new Document(
+  const updatedDocument = new Document(
     {
-      workItems: currentDocument.workItems,
       ...entryMetadata,
+      createdAt: currentDocument.createdAt, // eslint-disable-line
       documentId,
       documentType,
       relationship: 'primaryDocument',
       userId: user.userId,
+      workItems: currentDocument.workItems,
+      ...caseEntity.getCaseContacts({
+        contactPrimary: true,
+        contactSecondary: true,
+      }),
     },
     { applicationContext },
   ).validate();
 
-  documentEntity.generateFiledBy(caseToUpdate);
-  documentEntity.setQCed(user);
+  updatedDocument.generateFiledBy(caseToUpdate, true);
```

Comment:
why is this one force-generated?

Reply:
The original document already had a `filedBy`, and the new document generated (or _updated_ for terminology) needed to have its `filedBy` updated. The force flag was introduced since it doesn't update the `filedBy` by default if it already existed.

Although, I do believe this needs to have a unit test added if I'm not mistaken.

Reply:
Nice catch. Did the unit test get added in the updates? I'm not seeing it.

Reply:
It seems it was already added -- either someone caught it and took it upon themselves, or I forgot I did it.

Reply:
This has been merged in! 😄

Reply:
🎉 thanks!